app/src/org/commcare/google/services/ads/AdMobManager.java

```diff
@@ -0,0 +1,118 @@
+package org.commcare.google.services.ads;
+
+import android.content.Context;
+import android.view.View;
+import android.widget.FrameLayout;
+
+import com.google.android.gms.ads.AdRequest;
+import com.google.android.gms.ads.AdSize;
+import com.google.android.gms.ads.AdView;
+import com.google.android.gms.ads.MobileAds;
+
+import org.commcare.CommCareApplication;
+import org.commcare.dalvik.BuildConfig;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Aliza Stone
+ */
+public class AdMobManager {
+
+    private static final String AD_TYPE_BANNER = "banner";
+    private static final String AD_TYPE_INTERSTITIAL = "interstitial";
+
+    private static final Map<String, String> adMobIdsMap = initAdMobIdsMap();
+
+    private static final String CLINICAL_SCALES_PACKAGE_ID =
+            "org.commcare.consumerapps.clinicalscales";
+
+    public static Map<String, String> initAdMobIdsMap() {
+        Map<String, String> map = new HashMap<>();
+        map.put(CLINICAL_SCALES_PACKAGE_ID, "ca-app-pub-8038725004530429~5040587593");
+        return map;
+    }
+
+    public static void requestBannerAdForView(Context context, FrameLayout adContainer) {
+        String currentConsumerAppPackageId = getPackageIdentifierForCurrentConsumerApp(context);
+        if (hasAdmobId(currentConsumerAppPackageId)) {
+            AdView adView = buildBannerAdView(context, currentConsumerAppPackageId);
+            adContainer.setVisibility(View.VISIBLE);
+            adContainer.addView(adView);
+
+            AdRequest adRequest = buildAdRequest();
+            adView.loadAd(adRequest);
+        } else {
+            adContainer.setVisibility(View.GONE);
+        }
+    }
+
+    private static AdView buildBannerAdView(Context context, String currentConsumerAppPackageId) {
+        AdView adView = new AdView(context);
+        adView.setAdSize(AdSize.BANNER);
+        adView.setAdUnitId(getBannerAdUnitIdForCurrentConsumerApp(currentConsumerAppPackageId));
+        return adView;
+    }
+
+    private static AdRequest buildAdRequest() {
+        if (BuildConfig.DEBUG) {
```

Comment:
Sort of. This will still load an ad, but it will load a "test" ad. The AdMob terms of service are very strict about the fact that you should not be loading any real ads while debugging (I assume because it would inflate people's viewing numbers). So they provide some pre-made test ads that you can request instead.